gitlab/resource_gitlab_service_pipelines_email.go

```diff
@@ -0,0 +1,103 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabServicePipelinesEmail() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabServicePipelinesEmailCreate,
+		Read:   resourceGitlabServicePipelinesEmailRead,
+		Update: resourceGitlabServicePipelinesEmailCreate,
+		Delete: resourceGitlabServicePipelinesEmailDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceGitlabServicePipelinesEmailImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"recipients": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"notify_only_broken_pipelines": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"branches_to_be_notified": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"all", "default", "protected", "default_and_protected"}, true),
+				Default:      "default",
+			},
+		},
+	}
+}
+
+func resourceGitlabServicePipelinesEmailSetToState(d *schema.ResourceData, service *gitlab.PipelinesEmailService) {
+	d.SetId(fmt.Sprintf("%d", service.ID))
```

Comment:
There would be a bug here, if you're using the project ID as the resource ID

Reply:
Let's get it straight because I am not sure we understand each other:

1) The project ID is different from the resource ID (== service ID of the GitLab API in my current design)
2) I cannot read the service from the GitLab API with the service ID (I need the project ID)
3) I don't think it is a good idea to use the project ID as the resource ID, but may be it is usual to do that?

Reply:
Ah, I thought you had already settled on using the project ID as the resource ID. My bad.

I would definitely suggest using the project ID as the resource ID, because it is the ID that you use to import, and more importantly, the ID you use to lookup the service in GitLab.

Reply:
OK let's fo that than. Thanks